kitchen/bin/waiter-init

```diff
@@ -47,6 +47,15 @@ handle_k8s_terminate() {
 }
 trap handle_k8s_terminate SIGTERM
 
+# Track container restart count
+waiter_restart_count=$(( $([ -f .waiter-container-runs ] && cat .waiter-container-runs) ))
+echo $(( $waiter_restart_count + 1 )) > .waiter-container-runs
+
+# Ensure that HOME is set to the fresh working directory for this container instance.
+export HOME="$PWD/r${waiter_restart_count}"
+mkdir -p "$HOME"
+cd "$HOME"
```

Comment:
Yes, this breaks values like `MESOS_SANDBOX`. Good catch.

We discussed this offline, and decided to add a `latest` symlink (a la mesos) to point to the latest working directory. This will simplify the environment setup, and lets us avoid overriding any such environment variables in this script.